mage_ai/orchestration/db/models/oauth.py

```diff
@@ -188,6 +188,18 @@ def create_default_roles(
                     ],
                     commit=False,
                 )
+            elif permissions:
+                role.update(
+                    permissions=role.permissions + [
```

Comment:
Can this role already have the permission?

Reply:
Can you add comments to this method?

Reply:
why? does this method always return permissions? https://github.com/mage-ai/mage-ai/blob/master/mage_ai/orchestration/db/models/oauth.py#L277-L299

Reply:
it doesn't always return permissions. it will first check to see if permissions already exist for the given entity and entity id, and only create new permissions if they don't exist.

i updated the method to be a little more clear. i'll add some comments too

Reply:
OK, if we want to reduce the unnecessary queries, we can just get the permission from the permissions list instead of making individual filter queries.